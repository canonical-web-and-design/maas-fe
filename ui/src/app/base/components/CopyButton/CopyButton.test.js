import { mount } from "enzyme";
import React from "react";

import CopyButton from "./CopyButton";

describe("CopyButton ", () => {
  let execCommand;

  beforeEach(() => {
    execCommand = document.execCommand;
    document.execCommand = jest.fn();
  });

  afterEach(() => {
    document.execCommand = execCommand;
  });

  it("can render", () => {
    const wrapper = mount(<CopyButton value="Test key" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("can copy a value", () => {
    const wrapper = mount(<CopyButton value="Test key" />);
    wrapper.find("Button").simulate("click");
    expect(document.execCommand).toHaveBeenCalled();
    expect(wrapper.find("input").is(":focus")).toBe(true);
  });
});
