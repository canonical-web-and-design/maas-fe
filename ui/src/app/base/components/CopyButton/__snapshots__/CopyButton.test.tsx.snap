// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyButton can render 1`] = `
<CopyButton
  value="Test key"
>
  <Button
    appearance="base"
    className="is-dense u-table-cell-padding-overlap u-no-margin--right"
    hasIcon={true}
    onClick={[Function]}
  >
    <button
      className="p-button--base has-icon is-dense u-table-cell-padding-overlap u-no-margin--right"
      onClick={[Function]}
    >
      <i
        className="p-icon--copy"
      >
        Copy
      </i>
    </button>
  </Button>
  <input
    type="hidden"
    value="Test key"
  />
</CopyButton>
`;
